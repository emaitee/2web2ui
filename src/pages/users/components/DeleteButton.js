import React from 'react';
import _ from 'lodash';
import { Button } from '@sparkpost/matchbox';

export default function DeleteButton({ disabled = false, name = null, onClick = _.noop }) {
  // Display nothing when disabled
  if (disabled) { return null; }

  const handleClick = (event) => { onClick(name); };

  return (
    <Button name={name} onClick={handleClick}>
      Delete
    </Button>
  );
}
