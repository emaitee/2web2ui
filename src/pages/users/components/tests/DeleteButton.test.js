import { shallow } from 'enzyme';
import React from 'react';
import { Button } from '@sparkpost/matchbox';
import DeleteButton from '../DeleteButton';

describe('Component: User Page DeleteButton', () => {

  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      name: 'delete-button-name'
    };
    wrapper = shallow(<DeleteButton {...props} />);
  });

  it('should render correctly by default', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when disabled', () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle a click', () => {
    const e = {};
    const onClick = jest.fn();
    wrapper.setProps({ onClick });
    wrapper.find(Button).simulate('click', e);
    expect(onClick).toHaveBeenCalledWith('delete-button-name');
  });

});
