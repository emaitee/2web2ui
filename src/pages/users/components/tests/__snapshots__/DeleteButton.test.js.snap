// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: User Page DeleteButton should render correctly by default 1`] = `
<Button
  name="delete-button-name"
  onClick={[Function]}
  size="default"
>
  Delete
</Button>
`;

exports[`Component: User Page DeleteButton should render when disabled 1`] = `""`;
